synapse/lib/lmdb.py

```diff
@@ -553,3 +565,411 @@ def encodeValAsKey(v, isprefix=False):
             return _STR_VAL_MARKER + v_enc
         else:
             return _STR_VAL_MARKER + v_enc + b'\x00'
+
+class Slab(s_base.Base):
+    '''
+    A "monolithic" LMDB instance for use in a asyncio loop thread.
+    '''
+    COMMIT_PERIOD = 1.0  # time between commits
+
+    async def __anit__(self, path, **opts):
+        await s_base.Base.__anit__(self)
+
+        self.path = path
+        self.optspath = os.path.join(path, 'opts.json')
```

Comment:
this feels incorrect.  in `010` we pretty much moved to using yaml for all configuration files; and using the config helper from common.py to parse those out.  This has a different voice from that approach.

Reply:
todo